### ISSUE TITLE
Compiled Code Snippets

### ISSUE THREAD

**Issue by GrafBlutwurst** [rabbitonweb/frost#11]
Like tut or babbel. Allow for code snippets to be compiled against a project so they don't get stale.

**Comment by rabbitonweb** [comment]
@GrafBlutwurst Can I assume that for now you require integration with `tut`? Or you would like to have full `tut` experience in frost? Honestly I would rather have a plugin that integrates nicely with `tut`. 

Let's say you have a `tut` example in `src/tut/examples.md` that looks like

````
Here is how you add numbers:
```tut
1 + 1
```
````


The plugin could look like

````
# Scala Example
```frost:tut
path=src/tut/examples.md
```
````

and then the plugin would 
1. run `sbt tutOnly path`
2. substitute the `tut` output with plugin content

So that finally documentation generated by `frost` would be

````
# Scala Example
Here is how you add numbers:
```scala
scala> 1 + 1
res0: Int = 2
```
````

Would that work?

**Comment by GrafBlutwurst** [comment]
yeah integration with `tut` like this would be cool. What other tools would there be for other languages? Maybe emacs' babel can be an inspiration here?
